pkg/apiserver/debugapi/endpoint_def.go

```diff
@@ -0,0 +1,537 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debugapi
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pingcap/kvproto/pkg/metapb"
+
+	"github.com/pingcap/tidb-dashboard/pkg/apiserver/debugapi/endpoint"
+	"github.com/pingcap/tidb-dashboard/pkg/apiserver/model"
+)
+
+var pprofKindsParam = &endpoint.APIParam{
+	Model: APIParamModelEnum([]EnumItem{
+		{Value: "allocs"},
+		{Value: "block"},
+		{Value: "cmdline"},
+		{Value: "goroutine"},
+		{Value: "heap"},
+		{Value: "mutex"},
+		{Value: "profile"},
+		{Value: "threadcreate"},
+		{Value: "trace"},
+	}),
+	Name: "kind", Required: true,
+}
+
+var pprofSecondsParam = &endpoint.APIParam{
+	Model: APIParamModelEnum([]EnumItem{
+		{Name: "10s", Value: "10"},
+		{Name: "30s", Value: "30"},
+		{Name: "60s", Value: "60"},
+		{Name: "120s", Value: "120"},
+	}),
+	Name: "seconds",
+}
+
+func timeoutMiddleware(req *endpoint.Request, sec string) error {
+	i, err := strconv.ParseInt(sec, 10, 64)
+	if err != nil {
+		return err
+	}
+	duration := time.Duration(i) * time.Second
+	req.Timeout = duration + duration/2
+	return nil
+}
+
+func registerEndpoint(c *endpoint.Client) {
```

Comment:
Seems that this function can be eliminated and can be moved to `newClient()`, since it is not necessary for us to register multiple different endpoints (as endpoints are static):

```go
func newClient(..., endpoints: []endpoint.APIModel) {
    // deal with the endpoints, build hashmap for it.
}
```